### PR TITLE
Devel

### PR DESCRIPTION
**Pull request by fvalenza**
Keep track of the bodies as Frames

### CODE REVIEW

**Comment by nmansard** [comment]
Thanks for this PR. Unification of names with "frames" is good. Should we start to document body listing as deprecated? 
What is the interest of remembering the type of frames (body vs oppoint vs joint)? Is there any explicit demand for this feature? If so, it would be nice to document this demand somewhere to keep track of it.

**Comment by jcarpent** [comment]
This is related to #179.

File: src/spatial/frame.hpp [hunk 1]
```diff
@@ -29,26 +29,35 @@
 namespace se3
 {
 
+  enum FrameType
+  {
+    OP_POINT,
```
Comment (jcarpent) [inline src/spatial/frame.hpp]: why not to use OP_FRAME instead of OP_POINT?

**Comment by jcarpent** [comment]
I would go a little bit deeper by adding the type Joint inside `FrameType` and registering all the joints inside. The idea is to have everything directly accessible from the vector frames in Model, simplifying the API.

**Comment by nmansard** [comment]
This is an interesting proposition. I am afraid that it may simplify the API a little bit too much (loosing efficiency in runtime). I think we should try to imagine how this "frames" could be used by a third-party.

**Comment by jcarpent** [comment]
The adding of Joint in the frame list is just a redundant information to help the end-user to easily retrieve the objects (SENSOR, BODY, JOINT, ...) created during parsing procedure.
The efficiency can be near the same if in the method `getFrameJacobian` we make the distinction between Joint frame and the other frame types.

**Comment by jmirabel** [comment]
Having a unified way of getting joints regardless of their type (Fixed or not) is a feature that will simplify the work of porting HPP to Pinocchio.

**Comment by jcarpent** [comment]
:+1:

**Pull request merged**